packages/near-membrane-base/src/membrane.ts

```diff
@@ -190,1428 +189,1467 @@ export function init(
+                    targetPointer();
+                    const target = getSelectedTarget();
+                    let targetIntegrityTraits = TargetIntegrityTraits.None;
+                    try {
+                        // a revoked proxy will break the membrane when reading the meta
+                        if (ObjectIsFrozen(target)) {
+                            targetIntegrityTraits |=
+                                TargetIntegrityTraits.IsSealed &
+                                TargetIntegrityTraits.IsFrozen &
+                                TargetIntegrityTraits.IsNotExtensible;
+                        } else if (ObjectIsSealed(target)) {
+                            targetIntegrityTraits |= TargetIntegrityTraits.IsSealed;
+                        } else if (!ReflectIsExtensible(target)) {
+                            targetIntegrityTraits |= TargetIntegrityTraits.IsNotExtensible;
+                        }
+                        // if the target was revoked or become revoked during the extraction
+                        // of the metadata, we mark it as broken in the catch.
+                        isArrayOrNotOrThrowForRevoked(target);
+                    } catch {
+                        // intentionally swallowing the error because this method is just
+                        // extracting the metadata in a way that it should always succeed
+                        // except for the cases in which the target is a proxy that is
+                        // either revoked or has some logic that is incompatible with the
+                        // membrane, in which case we will just create the proxy for the
+                        // membrane but revoke it right after to prevent any leakage.
+                        targetIntegrityTraits |= TargetIntegrityTraits.Revoked;
+                    }
+                    return targetIntegrityTraits;
+                },
                 'callableGetTargetIntegrityTraits',
-                OUTBOUND_INSTRUMENTATION_LABEL
-            )
-        );
-        foreignCallableHasOwnProperty = foreignErrorControl(
+                INBOUND_INSTRUMENTATION_LABEL
+            ),
+            // callableHasOwnProperty
             instrumentCallableWrapper(
-                callableHasOwnProperty,
+                (targetPointer: Pointer, key: PropertyKey): boolean => {
+                    targetPointer();
+                    const target = getSelectedTarget();
+                    try {
+                        return ReflectApply(ObjectProtoHasOwnProperty, target, [key]);
+                    } catch (e) {
+                        throw pushErrorAcrossBoundary(e);
+                    }
+                },
                 'callableHasOwnProperty',
-                OUTBOUND_INSTRUMENTATION_LABEL
+                INBOUND_INSTRUMENTATION_LABEL
             )
         );
+        return (...hooks: Parameters<HooksCallback>) => {
```

Comment:
Maybe instead of `init()` this is called something like `createHooksCallback()`

Reply:
changed!